File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/dialogue/config/DomainServiceConfiguration.java
package com.study.webflux.rag.infrastructure.dialogue.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.study.webflux.rag.domain.dialogue.service.SentenceAssembler;

/** 프레임워크 비의존 도메인 서비스 빈을 등록합니다. */
@Configuration
public class DomainServiceConfiguration {

	@Bean
	public SentenceAssembler sentenceAssembler() {
		return new SentenceAssembler();
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/dialogue/config/DomainServiceConfiguration.kt
package com.miyou.app.infrastructure.dialogue.config

import com.miyou.app.domain.dialogue.service.SentenceAssembler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class DomainServiceConfiguration {
    @Bean
    fun sentenceAssembler(): SentenceAssembler = SentenceAssembler()
}
